core/services/fluxmonitor/flux_monitor.go

```diff
@@ -878,3 +884,44 @@ func OutsideDeviation(curAnswer, nextAnswer decimal.Decimal, threshold float64)
 	logger.Infow("Deviation threshold met", loggerFields...)
 	return true
 }
+
+const pathologicalSkew = 30 * time.Second
+
+func MakeNewRoundIdleTimer(log *contracts.LogNewRound, idleThreshold models.Duration, clock Clock) <-chan time.Time {
+	timeNow := clock.Now()
+	var roundStarted time.Time
+	if log.StartedAt.IsInt64() {
+		roundStarted = time.Unix(log.StartedAt.Int64(), 0)
+	} else {
+		logger.Errorf("value for log.StartedAt %s would overflow int64, ignoring system clock and setting idle timer to max duration of %s", log.StartedAt.String(), idleThreshold.Duration())
+		return clock.After(idleThreshold.Duration())
+	}
+	if roundStarted.After(timeNow) {
+		logger.Warnf("round started time of %s is later than current system time of %s, ignoring system clock and setting idle timer to max duration of %s. This machine probably has a slow clock", roundStarted.String(), timeNow.String(), idleThreshold.Duration().String())
```

Comment:
As in the previous case: aren't we ignoring the Ethereum clock rather than the system clock?